Fix grammar across docs content (#519)
icerpc/icerpc-docs: content/slice/language-reference/index.md

- A preprocessing language for conditional compilation
- A language for describing definitions with doc comments

> Each of these sub-languages is in turn specified by a set of two [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar):

Reviewer: I think the real issue wasn't the 'is', but using 'each' when we only had 2 things.
```suggestion
Both of these sub-languages are in turn specified by a set of two [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar):
```

Author: Held off on this one: the intro two lines up says "Slice consists of three sub-languages" (core, preprocessing, doc comments), and each of the three has its own lexical+syntactic grammar pair (preprocessor-directives.md and doc-comments.md both define theirs). So "Each" refers to those three. If you'd rather count two, line 9 would need updating too — happy to apply whichever pair you prefer.
